dashboard/src/pages/ingest/upload/index.vue

```diff
@@ -0,0 +1,54 @@
+<script setup lang="ts">
+import Uppy from "@uppy/core";
+import { Dashboard } from "@uppy/vue";
+import XHR from "@uppy/xhr-upload";
+import { useRouter } from "vue-router/auto";
+
+import { getPath } from "@/client";
+import { useAuthStore } from "@/stores/auth";
+import { useLayoutStore } from "@/stores/layout";
+import IconUpload from "~icons/clarity/plus-circle-line";
+
+import "@uppy/core/dist/style.min.css";
+import "@uppy/dashboard/dist/style.css";
+import "@uppy/progress-bar/dist/style.min.css";
+
+const router = useRouter();
+const authStore = useAuthStore();
+const layoutStore = useLayoutStore();
+
+const GB = 1000 ^ 3;
+const maxFileSize = 500 * GB;
+
+layoutStore.updateBreadcrumb([{ text: "Ingest" }, { text: "Upload SIPs" }]);
+
+const uppy = new Uppy({ restrictions: { maxFileSize: maxFileSize } }).use(XHR, {
+  endpoint: getPath() + "/ingest/sips/upload",
+  headers: { Authorization: `Bearer ${authStore.getUserAccessToken}` },
+  allowedMetaFields: false,
+  getResponseData: () => {
+    return { url: "" };
+  },
+  onAfterResponse: (xhr) => {
+    if (xhr.status === 201) {
+      setTimeout(() => {
+        router.push({
+          path: "/ingest/sips",
+        });
+      }, 500);
+    }
+  },
+});
```

Comment:
Do we want to do something like what they suggest to refresh the access token if needed?

https://uppy.io/docs/xhr-upload/#how-can-i-refresh-auth-tokens-after-they-expire

We sign out the user directly in other requests:

https://github.com/artefactual-sdps/enduro/blob/main/dashboard/src/client.ts#L68-L73

Not sure what would happen here. Would it should an error? Would it continue retrying?

Reply:
I've implemented the `authStore.signinSilent()` method to refresh the access token if needed. This solution works for my tests instead of failing if the token is expired, as it used to do.  It's hard to manually test all the possible scenarios though, and I'm not sure how to do automated testing of how the uploader handles an expired token. :(

I tried to add a `catch()` to the token renewal process in case `authStore.signinSilent()` fails, but I didn't see any difference in the uploader behaviour, and again I'm not sure how to automate testing this functionality.

Reply:
Nice, thanks @djjuhasz! Looks like a good approach, I'd like to revisit how we do this in other API interactions, so we can go from this.

> I tried to add a catch() to the token renewal process in case authStore.signinSilent() fails, but I didn't see any difference in the uploader behaviour, and again I'm not sure how to automate testing this functionality.

I think the issue is that we don't return the promise from `authStore.signinSilent()`, I think we should deal with the error from the manager there, no need to do it now.